### PR TITLE
Adapt mongo connector to hadoop v1

### PR DESCRIPTION
**Pull request by xardazz**
Fix NaN handling

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -10,7 +10,7 @@ ext.hadoopBinaries = "${rootDir}/hadoop-binaries".toString()
 ext.javaDriverVersion = '3.2.1'
 ext.hiveVersion = '1.2.1'
 ext.pigVersion = '0.15.0'
-ext.hadoopVersion = '2.7.2'
+ext.hadoopVersion = '1.2.1'
```
Comment (motymichaely) [inline build.gradle:13]: @xardazz - Isn't it possible to provide gradle the hadoopVersion on compile time? This should be a variable, not a constant.. no?

Comment (xardazz) [inline build.gradle:13]: @motymichaely I'm not gradle  expert, but  I don't think so. You can move this property to external file, but I've never seen ways to specify it in command line. Also  v1 and v2 have  different libraries, so having version dynamic won't help

File: pig/src/main/java/com/mongodb/hadoop/pig/MongoLoader.java [hunk 2]
```diff
@@ -129,7 +129,7 @@ public void prepareToRead(final RecordReader reader, final PigSplit split) throw
                 boolean include = (Boolean) entry.getValue();
                 // Add the name of the outer-level field if this is a nested
                 // field. Pig will take care of pulling out the inner field.
-                String key = StringUtils.split(entry.getKey(), '.')[0];
+                String key = StringUtils.split(entry.getKey(), '\\', '.')[0];
```
Comment (motymichaely) [inline pig/src/main/java/com/mongodb/hadoop/pig/MongoLoader.java:132]: @xardazz What's this for?

Comment (xardazz) [inline pig/src/main/java/com/mongodb/hadoop/pig/MongoLoader.java:132]: @motymichaely In hadoop v1 there is no method with 2 parameters.  In hadoop v2 it appears having '\\' passed to the 3-parameter method.

**Pull request merged**